### DISCUSSION TITLE
How to locate problems by exception code -> CHANNEL_ERROR - expected 'channel.open'(40, 10)

### DISCUSSION THREAD

**Discussion by kingfer30** [php-amqplib/php-amqplib#931] [Q&A]
My project uses a queue to connect MQ push messages in the queue. The queue is a resident process (it will restart regularly), this exception is occasionally found, and the returned code (40, 10) cannot be read. How to locate the problem?

Stack information :
【异常原因】：CHANNEL_ERROR - expected 'channel.open'(40, 10)
【异常堆栈】：
#0 [internal function]: PhpAmqpLib\Connection\AbstractConnection->connection_close(Object(PhpAmqpLib\Wire\AMQPReader))
#1 /mnt/data/dhwms/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(216): call_user_func(Array, Object(PhpAmqpLib\Wire\AMQPReader)) 
#2 /mnt/data/dhwms/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(373): PhpAmqpLib\Channel\AbstractChannel->dispatch('10,50', '\x01\xF8'CHANNEL_ERRO...', NULL) 
#3 /mnt/data/dhwms/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(675): PhpAmqpLib\Channel\AbstractChannel->wait() 
#4 /mnt/data/dhwms/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(234): PhpAmqpLib\Connection\AbstractConnection->wait_channel(1, 0) 
#5 /mnt/data/dhwms/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(351): PhpAmqpLib\Channel\AbstractChannel->next_frame(0) 
#6 /mnt/data/dhwms/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AMQPChannel.php(394): PhpAmqpLib\Channel\AbstractChannel->wait(Array, false, 0) 
#7 /mnt/data/dhwms/vendor/topthink/think-queue/src/queue/connector/Amqp.php(121): PhpAmqpLib\Channel\AMQPChannel->exchange_declare('exchange_sync_d...', 'direct', false, true, false) 
#8 /mnt/data/dhwms/vendor/topthink/think-queue/src/queue/connector/Amqp.php(106): think\queue\connector\Amqp->pushRaw('{"job":"\\app\\...', '01@sync_data') 

`
public function pushRaw($payload, $queue = null, array $options = [])
    {
        $queue_name = $this->getQueue($queue);

        $exchange = $this->getExchange($queue);

        $route = 'route_' . $queue_name;

        $topic_name = $exchange['exchange_name'];

        **//An exception was thrown here**
        $this->channel->exchange_declare($topic_name, $exchange['type'], false, true, false);

        $this->channel->queue_declare($queue_name, false, true, false, false);

        $this->channel->queue_bind($queue_name, $topic_name, $route);

        $msg = new AMQPMessage($payload, ["message_id" => json_decode($payload, true)['id'], 'content_type' => 'application/json', 'delivery_mode' => 2]);

        //发布消息
        $this->channel->basic_publish($msg, $topic_name, $route);

        return json_decode($payload, true)['id'] ?? null;
    }

`

**Comment by ramunasd** [comment]
Moving to discussions. Please ask for help there or in mailing lists.
